03/app.js

```diff
@@ -1,58 +1,71 @@
 const books = {
-    '978-83-7278-000-3': {
-        author: 'J.K. Rowling',
-        title: {
-            en: 'Harry Potter and the Philosopher\'s Stone',
-            pl: 'Harry Potter i Kamień Filozoficzny',
-        },
-        translator: {
-            en: null,
-            pl: 'Andrzej Polkowski',
-        }
+  "978-83-7278-000-3": {
+    author: "J.K. Rowling",
+    title: {
+      en: "Harry Potter and the Philosopher's Stone",
+      pl: "Harry Potter i Kamień Filozoficzny",
     },
-    '83-7278-007-2': {
-        author: 'J.K. Rowling',
-        title: {
-            en: 'Harry Potter and the Chamber of Secrets',
-            pl: 'Harry Potter i Komnata Tajemnic ',
-        },
-        translator: {
-            en: null,
-            pl: 'Andrzej Polkowski',
-        }
+    translator: {
+      en: null,
+      pl: "Andrzej Polkowski",
     },
-}
-
-books.getAuthor = function(isbn) {
-    // w tym przypadku this === books
-    
-    if(typeof this[isbn] === 'undefined') {
-        // taki ISBN nie istnieje w moim spisie
-        return null;
-    }
-
-    const author = this[isbn]['author'];
-    if(author) {
-        // zwracam informacje o autorze
-        return author;
-    }
+  },
+  "83-7278-007-2": {
+    author: "J.K. Rowling",
+    title: {
+      en: "Harry Potter and the Chamber of Secrets",
+      pl: "Harry Potter i Komnata Tajemnic ",
+    },
+    translator: {
+      en: null,
+      pl: "Andrzej Polkowski",
+    },
+  },
+};
 
-    // brak informacji o autorze
-    return false;
-}
+books.getAuthor = function (isbn) {
+  // w tym przypadku this === books
 
-books.getTitle = function(isbn, lang) {
+  if (typeof this[isbn] === "undefined") {
+    // taki ISBN nie istnieje w moim spisie
+    return null;
+  }
 
-}
+  const author = this[isbn]["author"];
+  if (author) {
+    // zwracam informacje o autorze
+    return author;
+  }
 
-books.getTranlator = function(isbn, lang) {
+  // brak informacji o autorze
+  return false;
+};
 
-}
+books.getTitle = function (isbn, lang) {
+  if (typeof this[isbn].title[lang] === "undefined") {
+    return null;
+  }
```

Comment:
Warto zapisywać sobie to do dodatkowej zmiennej (tak jak wyżej) i wtedy jest łatwiej.
Pamiętaj, że każdą wartość należy sprawdzić po kolei tj.
```
const bookByIsbn = this[isbn];
if(typeof bookByIsbn !== 'undefined') {
    const title = bookByIsb.title[lang];
    if(typeof title !== 'undefined') {
    }
}
```

lub odwrotnie, że najpierw sprawdzamy czy nie istnieje
```
const bookByIsbn = this[isbn];
if(typeof bookByIsbn === 'undefined') {
    return null;
}

const title = bookByIsb.title[lang];
if(typeof title === 'undefined') {
}
```